.github/workflows/ci_rust.yml

```diff
@@ -254,6 +254,7 @@ jobs:
       - name: Run cargo clippy
         run: |
           cargo clippy --manifest-path ${{env.ROOT_PATH}}/Cargo.toml --all-targets -- -D warnings
```

Comment:
Is it necessary to run on both "default features" and "all features"? My naive assumption is that "default features" would be a strict subset of all features.

Reply:
You need both. Even though features are additive, the code that those feature combinations gate doesn't have to be. For example, if you only test `--all-features`, then you miss the `not` branch:

```rust
#[cfg(feature = "my-feature")]
do_thing_a();

#[cfg(not(feature = "my-feature"))]
do_thing_b();
```